kb/deploy-new-version-of-workflow.md

```diff
@@ -0,0 +1,67 @@
+---
+slug: deploy-new-version-of-workflow
+title: Use Replay before deploy a new version of a Workflow
+tags:
+  - troubleshooting
+  - how-to
+date: 2023-05-11T00:00:00Z
+---
+
+When making code changes to a Workflow, it is important to avoid introducing non-deterministic behavior; otherwise, you may encounter non-determinism errors in your Workflow History.
+
+One way to prevent this error from occurring is to use the [Workflow Replayer](/workflows#replays) to replay existing Workflow Histories against your new code before deploying it to production.
+
+:::important
+
+While the Workflow Replay helps to avoid non-deterministic code, it does not guarantee that your code is deterministic.
+
+:::
+
+## How to use the Workflow Replayer
+
+To use the Workflow Replayer, you need to provide the following:
+
+- The Workflow Replayer to register the Workflow that is going to be replayed
+- A Workflow History to execute, either:
+  - Loaded directly from the Client
+  - Loaded from the JSON History file
+
+If you choose to load the Workflow History directly from the Client, consult the [SDK specific documentation](/application-development/testing#replay) for more information.
+
+If you chose to load the Event History from a JSON file on disk, use one the following methods:
```

Comment:
This says from JSON file on disk and then the first subsection is unrelated to JSON, files, or disk